tests/mnpe_test.py

```diff
@@ -18,20 +18,21 @@
 
 
 # toy simulator for continuous data with mixed parameters
-def mixed_param_simulator(theta: Tensor) -> Tensor:
+def mixed_simulator(theta, sigma2=0.1):
     """Simulator for continuous data with mixed parameters.
-
-    Args:
-        theta: Parameters with mixed types - continuous and discrete.
-    Returns:
-        x: Continuous observation.
-    """
+    Returns x~N(mu_z, tau^2)"""
     device = theta.device
 
-    # Extract parameters
-    a, b = theta[:, 0], theta[:, 1]
-    noise = 0.05 * torch.randn(a.shape, device=device).reshape(-1, 1)
-    return (a + 2 * b).reshape(-1, 1) + noise
+    if isinstance(sigma2, float):
+        sigma2 = torch.tensor(sigma2, device=device)
+
+    mu = theta[:, :2]
+    z = theta[:, -1]
+    # Select mu based on z
+    mu_z = torch.where(z == 0, mu[:, 0], mu[:, 1])
+    # Sample from N(mu_z, sigma2)
+    x = Normal(loc=mu_z, scale=torch.sqrt(sigma2)).rsample()
+    return x.reshape(-1, 1).to(device)
 
 
 @pytest.mark.gpu
```

Comment:
this seems to be failing because of the MixedIndependent prior not allowing the move to GPU. Thus, I would suggest to just mark those tests as `xfail` (plus, we can actually remove the "cpu" test because this is tested below anyways):

`@pytest.mark.parametrize("device", (pytest.param("gpu", marks=pytest.mark.xfail(raises=NotImplementedError, reason="MultipleIndependent does not support device handling yet"))`